runelite-client/src/main/java/net/runelite/client/plugins/banktags/tabs/TabInterface.java

```diff
@@ -322,12 +322,26 @@ public void handleClick(MenuOptionClicked event)
 		{
 			if (event.getMenuOption().startsWith(CHANGE_ICON))
```

Comment:
it would because iconTosSet is set to null after the if

Reply:
and then it would reselect the iconToSet later down

Reply:
I dont think so, where?

Reply:
https://github.com/runelite/runelite/blob/8c5226b1382142f8ec92ac9a5b1de7e8ce321365/runelite-client/src/main/java/net/runelite/client/plugins/banktags/tabs/TabInterface.java#L390

Reply:
Yeah, that is CHANGE_ICON and not CHANGE_ICON + "(" + tag + ")" what will the actual option be

Reply:
```
if (iconToSet != null)
		{
			if (event.getMenuOption().startsWith(CHANGE_ICON + "("))
			{
				event.consume();
				ItemComposition item = getItem(event.getActionParam());
				int itemId = itemManager.canonicalize(item.getId());
				iconToSet.setIconItemId(itemId);
				iconToSet.getIcon().setItemId(itemId);
				configManager.setConfiguration(CONFIG_GROUP, ICON_SEARCH + iconToSet.getTag(), itemId + "");
				
			}

			// Reset icon selection even when we do not clicked item with icon
			iconToSet = null;
		}
```

The iconToSet is nulled after the if then continues down the line and get set again. So it would perenually be set if you right click the same tab.

Reply:
That seems like correct behaviour, if you select change icon again

Reply:
I just tested it with fix I mentioned and it works perfectly

Reply:
Not if the intended behaviour would be to cancel the icon selection without having to click on another interface.

Reply:
I dont think that should be the intended behaviour, unless you would do it everywhere, e.g do early return if iconToSet != null

Reply:
How else to intuitively cancel selection without another widget?

Reply:
Imo clicking another widget is good enough. Cancelling selection by pressing same option what is even in right click menu is not very intuitive anyway

Reply:
Sure.